core/src/main/java/org/infinispan/util/concurrent/CommandAckCollector.java

```diff
@@ -115,12 +121,12 @@ public void createMultiKeyCollector(CommandInvocationId id, Collection<Address>
    /**
     * Acknowledges a {@link org.infinispan.commands.write.PutMapCommand} completion in the primary owner.
     *
-    * @param id          the {@link CommandInvocationId}.
+    * @param id          the id from {@link CommandInvocationId#getId()}.
     * @param from        the primary owner.
     * @param returnValue the return value.
     * @param topologyId  the topology id.
     */
-   public void multiKeyPrimaryAck(CommandInvocationId id, Address from, Map<Object, Object> returnValue,
+   public void multiKeyPrimaryAck(long id, Address from, Map<Object, Object> returnValue,
```

Comment:
Shouldn't this `id` be boxed as well?

Actually, I think I'd prefer keeping an unboxed id in the collectors, if only for the extra object header + reference.